hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/AzureBlobFileSystemStore.java

```diff
@@ -1314,7 +1352,30 @@ private String convertXmsPropertiesToCommaSeparatedString(final Hashtable<String
     return properties;
   }
 
+  private boolean isKeyForPrefixSet(String key, Set<String> dirSet) {
+
+    for (String dir : dirSet) {
+      if (dir.isEmpty() || key.startsWith(dir)) {
+        return true;
+      }
+
+      try {
+        URI uri = new URI(dir);
+        if (null == uri.getAuthority()) {
```

Comment:
nit: the 2 if here can be combined with &&

Reply:
code is removed.